internal/locate/region_request.go

```diff
@@ -2029,6 +2029,9 @@ func (s *RegionRequestSender) onSendFail(bo *retry.Backoffer, ctx *RPCContext, r
 }
 
 func isCauseByDeadlineExceeded(err error) bool {
+	if errConn, ok := err.(*client.ErrConn); ok {
+		err = errConn.Unwrap()
+	}
```

Comment:
Why not to implement the causer interface for `ErrConn`?

Reply:
Nice catch, implement the [causer](https://pkg.go.dev/github.com/pkg/errors#readme-retrieving-the-cause-of-an-error) is a better way to fix. Done.